browser/net/brave_ad_block_tp_network_delegate_helper.cc

```diff
@@ -57,10 +57,10 @@ content::WebContents* GetWebContents(int render_process_id,
 void ShouldBlockAdOnTaskRunner(std::shared_ptr<BraveRequestInfo> ctx,
                                base::Optional<std::string> canonical_name) {
   bool did_match_exception = false;
-  std::string tab_host = ctx->tab_origin.host();
+  std::string source_host = ctx->initiator_url.host();
```

Comment:
`initiator_url` can be null for browser-initiated requests [1] - are you sure that this is what we need?

[1] https://source.chromium.org/chromium/chromium/src/+/master:services/network/public/mojom/url_loader.mojom;l=129?q=url_loader.mojom

Reply:
If a third-party frame initiates a request, we definitely need to capture the frame URL, not the tab URL. If there's a better way to do that, I'm all for it.

That being said, are there any instances of browser-initiated requests that might need adblocking? Otherwise I think we can just return early if `source_host == null`.

Reply:
We could add frame origin from `IsolationInfo` from `ResourceRequest` to `BraveRequestContext` and use it here... `initiator_url` also looks suitable, probably we should just leave it as is, but keep an eye on nightly/beta

Reply:
Cool, I updated to make sure it's a valid URL - will keep watching to make sure everything else still works as expected.